core/hosting/HostWithCoreClrHost/src/SampleHost.cpp

```diff
@@ -0,0 +1,324 @@
+        return -1;
+    }
+
+    if (createManagedDelegate == NULL)
+    {
+        printf("coreclr_create_delegate not found");
+        return -1;
+    }
+
+    if (shutdownCoreClr == NULL)
+    {
+        printf("coreclr_shutdown not found");
+        return -1;
+    }
+
+    //
+    // STEP 3: Construct AppDomain properties used when starting the runtime
+    //
+
+    // Construct the trusted platform assemblies (TPA) list
+    // This is the list of assemblies that .NET Core can load as
+    // trusted system assemblies (similar to the .NET Framework GAC).
+    // For this host (as with most), assemblies next to CoreCLR will 
+    // be included in the TPA list
+    std::string tpaList;
+    BuildTpaList(runtimePath, ".dll", tpaList);
+
+    // <Snippet3A>
+    // Define CoreCLR properties
+    const char* propertyKeys[] = {
+        "TRUSTED_PLATFORM_ASSEMBLIES",      // Trusted assemblies (like the GAC)
+        "APP_PATHS",                        // Directories to probe for application assemblies
+        // "APP_NI_PATHS",                     // Directories to probe for application native images (not used in this sample)
+        // "NATIVE_DLL_SEARCH_DIRECTORIES",    // Directories to probe for native dlls (not used in this sample)
+    };
+
+    const char* propertyValues[] = {
+        tpaList.c_str(),
+        runtimePath
+    };
+    // </Snippet3>
+
+    //
+    // STEP 4: Start the CoreCLR runtime
+    //
+
+    // <Snippet3>
+    void* hostHandle;
+    unsigned int domainId;
+
+    // This function both starts the .NET Core runtime and creates
+    // the default AppDomain
+    int hr = initializeCoreClr(
+                    runtimePath,        // AppDomain base path
+                    "SampleHost",       // AppDomain friendly name
+                    sizeof(propertyKeys) / sizeof(char*),   // Property count
+                    propertyKeys,       // Property names
+                    propertyValues,     // Property values
+                    &hostHandle,        // Host handle
+                    &domainId);         // AppDomain ID
+    // </Snippet4>
+
+    if (hr >= 0)
+    {
+        printf("CoreCLR started; AppDomain %d created\n", domainId);
+    }
+    else
+    {
+        printf("coreclr_initialize failed - status: 0x%08x\n", hr);
+        return -1;
+    }                  
+
+    //
+    // STEP 5: Create delegate to managed code
+    //
+
+    // <Snippet5>
+    doWork_ptr managedDelegate;    
+    hr = createManagedDelegate(
+            hostHandle, 
+            domainId,
+            "ManagedLibrary, Version=1.0.0.0",
+            "ManagedLibrary.ManagedWorker",
+            "DoWork",
+            (void**)&managedDelegate);
+    // </Snippet5>
+
+    if (hr >= 0)
+    {
+        printf("Managed delegate created\n");
+    }
+    else
+    {
+        printf("coreclr_create_delegate failed - status: 0x%08x\n", hr);
+        return -1;
+    }    
+
+    //
+    // STEP 5.2: Invoking managed delegate
```

Comment:
Where is step 5.1?

Reply:
Hmm.. good question :) Removed this as it's all part of 'Step 5' (calling into the managed library)